src/azure-cli/azure/cli/command_modules/sql/commands.py

```diff
@@ -776,6 +780,19 @@ def load_command_table(self, _):
         g.custom_show_command('show', 'server_trust_group_get')
         g.custom_command('list', 'server_trust_group_list')
 
+    ###############################################
+    #         sql server deleted-server           #
+    ###############################################
+
+    deleted_servers_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.sql.operations#DeletedServersOperations.{}',
+        client_factory=get_sql_deleted_servers_operations)
+
+    with self.command_group('sql server deleted-server', deleted_servers_operations,
+                            client_factory=get_sql_deleted_servers_operations) as g:
+        g.custom_show_command('show', 'deleted_server_show')
+        g.custom_command('list', 'deleted_server_list')
```

Comment:
How about `az sql server list-deleted/show-deleted`. There's no need to add another sub command group `deleted-server`.